WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift

```diff
@@ -0,0 +1,252 @@
+                                               value: "Food, snack, sweet",
+                                               image: UIImage.categoriesIcon,
+                                               isLoading: viewModel.isGeneratingDetails)
+                        Divider()
+                            .background(Color(.separator))
+                        TitleAndValueDetailRow(title: Localization.tags,
+                                               value: "yummy, candy, chocolate",
+                                               image: UIImage.tagsIcon,
+                                               isLoading: viewModel.isGeneratingDetails)
+                        Divider()
+                            .background(Color(.separator))
+                        TitleAndValueDetailRow(title: Localization.shipping,
+                                               value: "Weight: 1kg\nDimension: 15 x 10 x 3 cm",
+                                               image: UIImage.shippingImage,
+                                               isLoading: viewModel.isGeneratingDetails)
+                    }
+                    .cornerRadius(Layout.cornerRadius)
+                }
+
+                // Feedback banner
+                FeedbackView(title: Localization.feedbackQuestion,
+                             backgroundColor: .init(uiColor: .init(light: .withColorStudio(.wooCommercePurple, shade: .shade0),
+                                                                   dark: .tertiarySystemBackground)),
+                             onVote: { vote in
+                    viewModel.handleFeedback(vote)
+                })
+                .renderedIf(viewModel.isGeneratingDetails == false)
+            }
+            .padding(insets: Layout.insets)
+            .toolbar {
+                ToolbarItem(placement: .confirmationAction) {
+                    if viewModel.isGeneratingDetails {
+                        ActivityIndicator(isAnimating: .constant(true), style: .medium)
+                    } else {
+                        Button("Save as draft") {
+                            viewModel.saveProductAsDraft()
+                        }
+                    }
+                }
+            }
+            .onAppear {
+                viewModel.generateProductDetails()
+            }
+        }
+    }
+}
+
+// MARK: - Subtypes
+private extension ProductDetailPreviewView {
+    /// View to contain basic product details
+    struct BasicDetailRow: View {
+        let content: String?
+        let isLoading: Bool
+        let dummyText = ProductDetailPreviewView.Constants.dummyText
+
+        typealias Layout = ProductDetailPreviewView.Layout
+
+        var body: some View {
+            Text(content ?? dummyText)
+                .bodyStyle()
+                .multilineTextAlignment(.leading)
+                .padding(Layout.contentPadding)
+                .frame(maxWidth: .infinity, alignment: .leading)
```

Comment:
❓ For my learning: What is the purpose of using a `.frame` modifier here? Thank you!

Reply:
If the text is not long enough, I noticed that the frame's width does not fit the whole screen. I tried `.fixedSize` but it didn't work, so I had to use this.